modules/hcp-topology-spread-constraint.adoc

```diff
@@ -0,0 +1,54 @@
+// Module included in the following assemblies:
+//
+// * hosted_control_planes/hcp-manage/hcp-manage-virt.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="hcp-topology-spread-constraint_{context}"]
+= Spreading node pool VMs by using topologySpreadConstraint
+
+By default, KubeVirt virtual machines (VMs) created by a node pool are scheduled on any available nodes that have the capacity to run the VMs. By default, the `topologySpreadConstraint` constraint is set to schedule VMs on multiple nodes.
+
+In some scenarios, node pool VMs might run on the same node, which can cause availability issues. To avoid distribution of VMs on a single node, use the descheduler to continuously honour the `topologySpreadConstraint` constraint to spread VMs on multiple nodes.
+
+.Prerequisites
+
+* You installed the {descheduler-operator}. For more information, see "Installing the descheduler".
+
+.Procedure
+
+* Open the `KubeDescheduler` custom resource (CR) by entering the following command, and then modify the `KubeDescheduler` CR to use the `SoftTopologyAndDuplicates` and `LongLifecycle` profiles so that you maintain the `topologySpreadConstraint` constraint settings.
++
+The `KubeDescheduler` CR named `cluster` runs in the `openshift-kube-descheduler-operator` namespace.
++
+[source,terminal]
+----
+$ oc edit kubedescheduler cluster -n openshift-kube-descheduler-operator
+----
++
+.Example `KubeDescheduler` configuration
+[source,yaml]
+----
+apiVersion: operator.openshift.io/v1
+kind: KubeDescheduler
+metadata:
+  name: cluster
+  namespace: openshift-kube-descheduler-operator
+spec:
+  mode: Automatic
+  managementState: Managed
+  deschedulingIntervalSeconds: 60 // <1>
+  profiles:
+  - SoftTopologyAndDuplicates   // <2>
+  - EvictPodsWithPVC           // <3>
+  - EvictPodsWithLocalStorage // <4>
+  - LongLifecycle            // <5>
```

Comment:
I'd suggest:
```
apiVersion: operator.openshift.io/v1
kind: KubeDescheduler
metadata:
  name: cluster
  namespace: openshift-kube-descheduler-operator
spec:
  managementState: Managed
  deschedulingIntervalSeconds: 30
  mode: "Automatic"
  profiles:
    - DevKubeVirtRelieveAndMigrate
    - SoftTopologyAndDuplicates
  profileCustomizations:
    devEnableSoftTainter: true
    devDeviationThresholds: AsymmetricLow
    devActualUtilizationProfile: PrometheusCPUCombined
```

`DevKubeVirtRelieveAndMigrate` is an enhanced variant of `LongLifecycle` for the Kubevirt use case.
With that `EvictPodsWithPVC` and `EvictPodsWithLocalStorage` are implicitly enabled.

Reply:
✔️